src/commands/hop.tsx

```diff
@@ -1,18 +1,23 @@
 import ErrorDisplay from '../components/error-display.js';
 import GumptionItemComponent from '../components/gumption-item-component.js';
-import React, { useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import SelectInput from 'ink-select-input';
 import { Box, Text, useApp } from 'ink';
+import { CommandProps } from '../command-registry.js';
 import { useGit } from '../hooks/use-git.js';
 
-function Hop() {
+function Hop({ input }: CommandProps) {
     const git = useGit();
     const { exit } = useApp();
     const [allBranches, setAllBranches] = useState<string[]>([]);
-    const [currentBranch, setCurrentBranch] = useState('');
+    const [currentBranch, setCurrentBranch] = useState<string | undefined>(
+        undefined
+    );
     const [newBranch, setNewBranch] = useState<string | undefined>(undefined);
     const [error, setError] = useState<Error | undefined>(undefined);
 
+    const [maybeSearchTerm] = input;
```

Comment:
Not a big fan of the "maybe" variable naming. Would `hasSearchTerm = !!input` work better?

Reply:
I like the maybe naming because it doesn't create an extra boolean variable just to check if a variable is undefined or not (something we can derive from the falsey nature of the variable itself)

The `input` prop is an array of all strings passed into this command. This is standard between all commands and the highlighted line of code attempts to fetch the first item in this array (it will be undefined if not included by the user)

In this function we both need access to the search term if provided and we need access to whether or not the search term is defined.

I'm not a fan of splitting this into 2 variables, but for clarity, are you suggesting:
```js
const [searchTerm] = input
const hasSearchTerm = !!searchTerm
```
?